data/model/test/test_oauth.py

```
@@ -94,3 +94,197 @@ def test_get_token_assignment_for_client_id(initialized_db):
+        db_auth_provider = MockDatabaseAuthorizationProvider()
+
+        # Malicious: https://example.com/callback@attacker.com
+        malicious_uri = f"{application.redirect_uri}@attacker.com"
+        assert not db_auth_provider.validate_redirect_uri(application.client_id, malicious_uri)
+
+        # Also block explicit username:password
+        malicious_uri_2 = "https://user:pass@attacker.com/callback"
+        assert not db_auth_provider.validate_redirect_uri(application.client_id, malicious_uri_2)
+
+
+def test_validate_redirect_uri_blocks_scheme_mismatch(initialized_db):
```

Ruff (0.14.5)

Unused function argument: `initialized_db`

(ARG001)